### ISSUE TITLE
Java 14 (possibly ZGC?) causes ticks to completely stop

### ISSUE THREAD

**Issue by TropheusJ** [jediminer543/JMT-MCMT#33]
attached are debug logs from java 8 and java 14 + the ZGC garbage collector. Java 8 appears to work fine, while 14 causes ticks to completely stop shortly after loading a world. once this happens, closing the world gets stuck on "saving world" and needs to be ended with task manager even though it's still responding. Possibly happens when generating fresh chunks, not sure. 
[java 8 debug log.txt](https://github.com/jediminer543/JMT-MCMT/files/5522194/java.8.debug.log.txt)
[java 14 debug log.txt](https://github.com/jediminer543/JMT-MCMT/files/5522196/java.14.debug.log.txt)
1.16.3 forge 34.1.42
mcmt 1.16.3-0.18.60

**Comment by jediminer543** [comment]
I've not tried it with Java 14 so IDK

Also sorry for delayed response, depression is a pain

**Comment by TropheusJ** [comment]
this occurs using any java version above 9. specifically from https://adoptopenjdk.net/ , but others should be no different. I can confirm that it even happens with https://github.com/ameisen/jdk-mc , a custom java 15 designed specifically for minecraft.

**Comment by TropheusJ** [comment]
actually, the issue is different now. not sure if its due to different mods, different version, or different JRE, but now worlds get stuck on 100% loading. the game does not freeze, it can be closed without issue. debug log attached from a test with java 9.
[debug.log](https://github.com/jediminer543/JMT-MCMT/files/5683856/debug.log)

**Comment by jediminer543** [comment]
Well that last log is SUPER wierd

> java.lang.NoClassDefFoundError: net/minecraftforge/event/entity/living/LivingEvent$LivingUpdateEvent

I.e. The requested class doesn't exist

This MIGHT be a class loader thing I guess; I'll add a thing to try fix that

Oh, also, as an additional note, I've been trying to get back to doing stuff and astral sorcery seems to cause some world loading issues on my setup; try disabling that

**Comment by TropheusJ** [comment]
disabled astral sorcery, same result. java 15. 
[debug.log](https://github.com/jediminer543/JMT-MCMT/files/5683887/debug.log)

**Comment by jediminer543** [comment]
Yeah, I didn't expect it to; I just don't know why a newer java version would cause the class loader to break

It's really odd

**Comment by ChromaPIE** [comment]
same for JDK 15

**Comment by jediminer543** [comment]
Hmm, there have been some other issues relating to java > 8; I need to spin up a test machine to see what causes it
